src/TestData/AdminSettings/Settings.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace GiveTestData\TestData\AdminSettings;
+
+use GiveTestData\Addon\View;
+use InvalidArgumentException;
+
+/**
+ * Example code to show how to add setting page to give settings.
+ *
+ * @package     GiveTestData\Addon
+ * @subpackage  Classes/Give_BP_Admin_Settings
+ * @copyright   Copyright (c) 2020, GiveWP
+ */
+class Settings extends \Give_Settings_Page {
+
+	/**
+	 * Settings constructor.
+	 *
+	 * @since 1.0.0
+	 */
+	public function __construct() {
+		$this->id          = 'give-test-data';
+		$this->label       = esc_html__( 'Test Data', 'give-test-data' );
+		$this->default_tab = 'forms';
+
+		parent::__construct();
+	}
+
+	/**
+	 * Get sections.
+	 *
+	 * @return array
+	 * @since 1.0.0
+	 */
+	public function get_sections() {
+		$sections = [
+			'forms'     => esc_html__( 'Generate Donation Forms', 'give-test-data' ),
+			'donors'    => esc_html__( 'Generate Donors', 'give-test-data' ),
+			'donations' => esc_html__( 'Generate Donations', 'give-test-data' ),
+			'pages'     => esc_html__( 'Generate Pages', 'give-test-data' ),
+		];
+
+		return apply_filters( 'give_get_sections_' . $this->id, $sections );
+	}
+
+	/**
+	 * Get sections pages.
+	 *
+	 * @return array
+	 * @since 1.0.0
+	 */
+	public function get_sections_pages() {
+		$actions = [
+			'forms'     => Forms::class,
+			'donors'    => Donors::class,
+			'donations' => Donations::class,
+			'pages'     => Pages::class,
+		];
+
+		return apply_filters( 'give_get_sections_pages_' . $this->id, $actions );
```

Comment:
@kjohnson any suggestions on how to "wire it up" directly?

Reply:
@kjohnson  just to clarify, my question was not about how to "wire up" the settings page directly, but on how to avoid making this application extensible. How to wire up multiple add-on domains in the future or remove existing ones? 
Using hooks and filters provides a quick and easy way of doing that. Is there a better way of doing this, I'm sure there is, but right now I'm not sure I'm aware of it. 

Would be great to discuss this further.
cc: @JasonTheAdams

Reply:
@kjohnson the problem here is that this helper class is extending the Give_Settings_Page class to register settings pages and pages sections. Writing new functionality to "wire up" the new settings pages directly won't be easy as Give_Settings_Page is not loaded through the container, and it's loaded very late. Maybe there is a way by using a static facade approach, but do we want to do that? I mean, just by mentioning this option, someone will say I'm cursing.

Reply:
I'm not entirely against Static Facades :smiley: 

Regarding my previous feedback, you make a good point about using the underlying `Give_Settings_Page` - let's stick with this and I'll reconsider. What you have certainly isn't a bad direction - it works and it makes sense :100:.